windows/CodePushResumeListener.cs

```diff
@@ -0,0 +1,49 @@
+﻿using ReactNative.Bridge;
+using System;
+
+namespace CodePush.ReactNative
+{
+    internal class CodePushResumeListener : ILifecycleEventListener
+    {
+        private DateTime? _lastSuspendDate = null;
+        private CodePushNativeModule _codePushNativeModule;
+
+        internal int MinimumBackgroundDuration { get; set; }
+
+        internal CodePushResumeListener(CodePushNativeModule codePushNativeModule, int minimumBackgroundDuration)
```

Comment:
What do you think about this constructor simply taking in an Action that it triggers as necessary? The consuming code could then just pass in a reference to the loadBundle method, and this class is completely simpllified to just listening to lifecycle events and throttling them as necessary.